protocol/x/affiliates/keeper/msg_server.go

```diff
@@ -18,6 +20,21 @@ func (k msgServer) RegisterAffiliate(ctx context.Context,
 
 func (k msgServer) UpdateAffiliateTiers(ctx context.Context,
 	msg *types.MsgUpdateAffiliateTiers) (*types.MsgUpdateAffiliateTiersResponse, error) {
+	sdkCtx := sdk.UnwrapSDKContext(ctx)
+
+	unconditionalRevShareConfig, err := k.revShareKeeper.GetUnconditionalRevShareConfigParams(sdkCtx)
+	if err != nil {
+		return nil, err
+	}
+	marketMapperRevShareParams := k.revShareKeeper.GetMarketMapperRevenueShareParams(sdkCtx)
+
+	if !k.revShareKeeper.ValidateRevShareSafety(*msg.Tiers, unconditionalRevShareConfig, marketMapperRevShareParams) {
```

Comment:
`msg.Tiers` should not be a pointer here. Looks like we forgot add `non nullable` option to `Tiers` on the message proto